action.yml

```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SSH to Google Cloud Platform compute instances
+author: Google LLC
+description: |-
+  Use this action to connect to Google Cloud Platform instances via ssh and execute user's commands.
+
+inputs:
+  instance_name:
+    description: |-
+      Name of the virtual machine instance to SSH into.
+    required: true
+
+  zone:
+    description: |-
+      Zone of the instance to connect to.
+    required: true
+
+  user:
+    description: |-
+      Specifies the username with which to SSH. 
+      If omitted, the user login name is used. If using OS Login, USER will be replaced by the OS Login user.
+    required: false
+
+  ssh_private_key:
+    description: |-
+      SSH private key with which to SSH.
+    required: true
+
+  ssh_public_key:
+    description: |-
+      SSH public key with which to SSH.
+    required: true
+
+  ssh_keys_folder:
+    description: |-
+      Path for a folder to store ssh keys
+    default: '/tmp/tmp-ssh-keys'
```

Comment:
The default shouldn't be hardcoded - it should be a randomly-generated path that we cleanup after the run. The path should be random each invocation.